### ISSUE TITLE
Track nullability of struct properties that are not auto-properties

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#29619]
There should be no warning reported for `s.P.ToString()` or `s.Q.ToString()`. Currently a warning is reported for `P`.
```c#
using System.Runtime.CompilerServices;
#nullable enable
struct S
{
    private object? _p;
    internal object? P
    {
        get { return _p; }
        set { _p = value; }
    }
    internal object? Q
    {
        get;
        set;
    }
    static void M(S s)
    {
        s.P = new object();
        s.Q = new object();
        s.P.ToString(); // warning: maybe null
        s.Q.ToString(); // no warning
    }
}
```

**Comment by gafter** [comment]
I don't know how we would "cutoff" the copying of state.  For example, consider this program

``` c#
struct S
{
    object? F;
    S P { set { this = value; } get { return this; } }
}
class Program
{
    void M(S s)
    {
        s.F = 2;
        for (int i = 0; i < 3; i++)
        {
            s.P = s;
        }
    }
}
```

This would cause the nullable walker not to terminate.  On the first pass it would compute the state of `S.F`.  Since the state changed, it would run a second pass and compute the state of `S.F.F`.  Since the state changed, it would run a second pass and compute the state of `S.F.F.F`.  And so on.

We would need a clear rule that cuts off this chain.
